Apps/Laboratory/src/Services/LaboratoryService.cs

```diff
@@ -86,224 +86,163 @@ public class LaboratoryService : ILaboratoryService
-            {
-                ResourcePayload = new(),
-                ResultStatus = ResultType.Error,
-                ResultError = UnauthorizedResultError(),
-            };
-
             string? accessToken = this.authenticationDelegate.FetchAuthenticatedUserToken();
-
-            if (accessToken != null)
+            if (accessToken == null)
             {
-                RequestResult<PhsaResult<PhsaLaboratorySummary>> delegateResult =
-                    await this.laboratoryDelegate.GetLaboratorySummary(hdid, accessToken).ConfigureAwait(true);
-
-                retVal.ResultStatus = delegateResult.ResultStatus;
-                retVal.ResultError = delegateResult.ResultError;
-                retVal.PageIndex = delegateResult.PageIndex;
-                retVal.PageSize = delegateResult.PageSize;
-                retVal.TotalResultCount = delegateResult.TotalResultCount;
+                return RequestResultFactory.Error<LaboratoryOrderResult>(UnauthorizedResultError());
+            }
 
-                PhsaLaboratorySummary? payload = delegateResult.ResourcePayload?.Result;
-                if (delegateResult.ResultStatus == ResultType.Success && payload != null)
-                {
-                    retVal.ResourcePayload.LaboratoryOrders =
-                        this.autoMapper.Map<IEnumerable<PhsaLaboratoryOrder>, IEnumerable<LaboratoryOrder>>(payload.LabOrders);
-                }
+            RequestResult<PhsaResult<PhsaLaboratorySummary>> delegateResult = await this.laboratoryDelegate.GetLaboratorySummary(hdid, accessToken).ConfigureAwait(true);
 
-                PhsaLoadState? loadState = delegateResult.ResourcePayload?.LoadState;
-                if (loadState != null)
-                {
-                    retVal.ResourcePayload.Queued = loadState.Queued;
-                    retVal.ResourcePayload.Loaded = !loadState.RefreshInProgress;
-                    if (loadState.RefreshInProgress)
+            PhsaLoadState? loadState = delegateResult.ResourcePayload?.LoadState;
+            if (loadState != null && loadState.RefreshInProgress)
+            {
+                return RequestResultFactory.ActionRequired(
+                    new LaboratoryOrderResult
                     {
-                        retVal.ResultStatus = ResultType.ActionRequired;
-                        retVal.ResultError = ErrorTranslator.ActionRequired("Refresh in progress", ActionType.Refresh);
-                        retVal.ResourcePayload.RetryIn = Math.Max(
-                            loadState.BackOffMilliseconds,
-                            this.labConfig.BackOffMilliseconds);
-                    }
-                }
+                        RetryIn = Math.Max(loadState.BackOffMilliseconds, this.labConfig.BackOffMilliseconds),
+                        Loaded = !loadState.RefreshInProgress,
+                    },
+                    ActionType.Refresh,
+                    "Refresh in progress");
             }
```

Comment:
The Loaded property also needs to be set when the request is successful.

The Queued property is no longer being populated at all.